### ISSUE TITLE
[90] Exploring Plain Vision Transformer Backbones for Object Detection

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#119]
[paper](https://arxiv.org/abs/2203.16527)

Object Detection 을 위해서 ViT 는 굳이 hierarchical 하게 갈 필요 없다.
hierarchical 하게 가지 않을 경우, pretrained model 을 사용하기에도 용이해진다.
논문 뒷편에서는 ViT 를 MAE 로 pretrain 하고 OD task 에 finetune 했을 때 성능까지 보여준다.

재밌는 점은 굳이 FPN 까지도 필요 없다는 점.
![image](https://user-images.githubusercontent.com/16400591/164719958-0e9e2875-84e5-49a7-a5fa-f07a068f2a37.png)

3가지 style 을 실험했는데,
(c) 만 해도 나쁘지 않다.
![image](https://user-images.githubusercontent.com/16400591/164720158-c919fbd4-c635-449e-a64d-ba53baf017ed.png)

backbone design 을 직접 했는데,
ViT-L 의 24 block 을 6개씩 4개의 block 으로 나누고,
각 block 마지막에 global self-attention 을 넣은 형태이다.

swin 처럼 sliding 도 사용하지 않는 형태다.
MHSA 를 사용하지 않으면 location 별로 정보가 모여있는 형태가 된다는 것 !
knowledge-propagation 을 위해 self-attention 이나 cnn 모듈을 사용하는데,
논문에서는 propagation 이라는 단어를 자주 사용하니 기억하도록 하자.

몇 가지 ablation 들이 재미있다.
![image](https://user-images.githubusercontent.com/16400591/164720605-382375c0-c28e-4ee7-848c-3e1b4ee4aabf.png)
(a) attention 은 conv 를 써서 window 를 쓰는 것도 꽤 괜찮게 성능이 나온다.
(b) naiive 는 3x3 하나, basic 은 3x3 2개를 사용. bottleneck 은 일반적인 resnet 형태이다.
(c) attention 은 고르게 분포시키는 게 좋다. 앞에 두는 것만은 피하자. 최악의 성능이다.
(d) propagation 은 4개면 충분하더라.

![image](https://user-images.githubusercontent.com/16400591/164721277-ba2687be-25a9-404f-a779-e9ed8abf7948.png)
속도까지 생각해서 저자들은 4개의 propagation을 사용했다.

MAE 까지 적용했는데,  21K pretrain 보다 성능이 잘 나오더라. (COCO, mask R-CNN)
![image](https://user-images.githubusercontent.com/16400591/164721524-005604d7-e42e-497e-ae05-1817df015335.png)

다른 백본들과 비교!
![image](https://user-images.githubusercontent.com/16400591/164721639-6b2720bc-968c-4942-9d83-bb65c6def6b3.png)
![image](https://user-images.githubusercontent.com/16400591/164721694-45e02d13-80b2-406e-9814-443c2c83a5fd.png)
![image](https://user-images.githubusercontent.com/16400591/164721788-8d526f9e-a575-467e-8629-ebf592fe052a.png)
